Simulator/src/main/kotlin/edu/drexel/se575/contract/Interpreter.kt

```diff
@@ -27,7 +32,7 @@ class Interpreter {
                     memory[instructions[index]?.get(1)!!.toInt()] = memory[instructions[index]?.get(2)!!.toInt()]
                 }
                 "2" -> { //trn
-                    // TODO Write an account value transfer function.
+                    transferAccountValue(accountList.toArray() as Array<Account?>, instructions[index]?.get(2), instructions[index]?.get(3), instructions[index]?.get(1)!!.toInt())
```

Comment:
This has type safety issues. Refer to the other times this "as Array<*>" syntax is used.

Reply:
Could replace with 
`accountList.toArray().filterIsInstance<Account?>().toTypedArray()`

Reply:
Or if you want to ensure the cast went correctly:
`accountList.toArray().filterIsInstance<Account?>().toTypedArray().apply { if (size != accountList.size) throw Exception() }`

Reply:
Well the worst they would be is null, their type is always defined explicitly. We solve that by using the ? nullable, so if I'm thinking correctly that should always be type-safe.

Reply:
> Or if you want to ensure the cast went correctly:
> `accountList.toArray().filterIsInstance<Account?>().toTypedArray().apply { if (size != accountList.size) throw Exception() }`

That's even better. We'll just want to keep track of all the unhandled exceptions, that could get ugly if we're not careful.

Reply:
The other option is to write a method that checks the cast. The accepted answer here was a pretty good explanation https://stackoverflow.com/questions/36569421/kotlin-how-to-work-with-list-casts-unchecked-cast-kotlin-collections-listkot

Reply:
Or just suppress it